fontir/src/ir.rs

```diff
@@ -1890,6 +1900,43 @@ pub struct Component {
     pub transform: Affine,
 }
 
+/// Data to inform construction of [CPAL](https://learn.microsoft.com/en-us/typography/opentype/spec/cpal#palette-table-header)
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub struct ColorPalettes {
+    // All palettes should have the same number of colors
+    pub palettes: Vec<Vec<Color>>,
+}
+
+impl ColorPalettes {
+    pub fn new(mut palettes: Vec<Vec<Color>>) -> Self {
+        // Trivial optimization: if there is only one palette we can deduplicate it
+        if palettes.len() == 1 {
+            palettes = vec![palettes
+                .into_iter()
+                .next()
+                .unwrap()
+                .into_iter()
+                .collect::<HashSet<_>>()
+                .into_iter()
+                .collect::<Vec<_>>()];
+        }
+        // Sort for stability in output
+        for p in palettes.iter_mut() {
+            p.sort();
+        }
+        Self { palettes }
+    }
+}
+
+/// We may in time need more than RGBA, but likely not for some years so start simple
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Color {
```

Comment:
we could just use the write-fonts `ColorRecord` type here instead of a newtype, if we wanted?